backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -68,6 +68,10 @@ internal val IrField.storageClass: FieldStorage get() {
     return when {
         descriptor.annotations.hasAnnotation(threadLocalAnnotationFqName) -> FieldStorage.THREAD_LOCAL
         descriptor.annotations.hasAnnotation(sharedAnnotationFqName) -> FieldStorage.SHARED
+        // TODO: simplify, once IR types are fully there.
+        // TODO: check if we do have backing field indeed.
+        !descriptor.isVar && type is IrSimpleType && ((type as IrSimpleType).
```

Comment:
`isVar` can (and must) be checked on `IrField` itself instead.

Reply:
By the way, how should we handle `@SharedImmutable var`?

Reply:
Yeah, shared immutable var shall be disabled - done.